### PR TITLE
Lookup the Gist API endpoint from the /api/v3 endpoint

### PR DESCRIPTION
**Pull request by simonmcc**
* AlfredGist v2.0.9 didn't work with GitHub Enterprise v2.13 as the
  correct API endpoint is https://{server}/api/v3/gists (the existing
  logic probably worked for previous versions of GHE)
* Instead of hardcoding or changing the gistconfig to store the Gist API
  we can look it up in a method that for both GitHub.com & GHE

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 EXTENSION = Gist.alfredworkflow
 BUNDLEID = pjkh.gist
-VERSION = 2.0.9
+VERSION = 2.0.10
 SOURCE_FILES = functions.sh gist.sh icon.png info.plist
 PKG_FILES = $(SOURCE_FILES)
 
-WORKFLOW_DIR = $(HOME)/Dropbox/AlfredApp/Alfred.alfredpreferences/workflows
+WORKFLOW_DIR = $(HOME)/Library/Application Support/Alfred 3/Alfred.alfredpreferences/workflows
```
Comment (phallstrom) [inline Makefile]: Would you mind reverting this change?

File: functions.sh [hunk 2]
```diff
@@ -236,6 +236,14 @@ function json_escape()
   echo -n "$1" | python -c 'import json,sys; print json.dumps(sys.stdin.read())'
 }
 
+#
+# Get Gist API endpoint from top level API
+#
+function get_gist_api()
+{
+  curl --silent --header "$auth_header" https://$server/api/v3 | python -c 'import json,sys; print json.loads(sys.stdin.read())["gists_url"].split("{")[0]'
```
Comment (phallstrom) [inline functions.sh]: For me this results in a call to `https://api.github.com/api/v3` which is 404. However, just plain `https://api.github.com/` returns a hash of urls that looks to be what you are after. Are you seeing something different? Does the latter URL work for you?

**Review by phallstrom — CHANGES_REQUESTED** [review]

File: functions.sh (hunk 2 above)
Comment (simonmcc) [reply functions.sh]: Ah, yeah, I'll see if I can make that a bit a smarter without having to change the config to store a url and not the server

**Comment by phallstrom** [comment]
Thanks!

**Comment by phallstrom** [comment]
new workflow is up at https://github.com/phallstrom/AlfredGist/blob/master/Gist.alfredworkflow?raw=true

**Pull request merged**